torch/fx/experimental/graph_gradual_typechecker.py

```diff
@@ -198,6 +200,53 @@ def bn2d_inference_rule(n: Node, op_type):
     else:
         raise TypeError(f'Cannot apply {op_type} with input type { arg_type} and existing type {n.type} on {n}')
 
+def calculate_hout(h_in, op_type):
+    if h_in == Dyn:
+        return Dyn
+
+    elif isinstance(h_in, int):
+        h_out = floor((h_in + (2 * op_type.padding[0] - op_type.dilation[0] *
```

Comment:
nit: i would factor out `calculate_hout` and `calculate_wout`, and parameterize by name/index, especially since Conv3d is the same exact logic for the fifth dimension.